server/src/main/java/org/elasticsearch/ingest/IngestService.java

```diff
@@ -1238,6 +1239,28 @@ private void executePipelines(
                         return; // document failed!
                     }
 
+                    for (StreamType streamType : StreamType.getEnabledStreamTypesForProject(project)) {
+                        if (streamType.matchesStreamPrefix(newIndex)
+                            && ingestDocument.getIndexHistory().contains(streamType.getStreamName()) == false) {
+                            exceptionHandler.accept(
+                                new IngestPipelineException(
+                                    pipelineId,
+                                    new IllegalArgumentException(
+                                        format(
+                                            "Pipelines can't re-route documents to child streams, but pipeline [%s] tried to reroute "
+                                                + "this document from index [%s] to index [%s]. Reroute history: %s",
```

Comment:
Nit: Wording - Does "child streams" make sense if I stumble onto this error message and I've never heard of the streams features? Should we mention "wired" or "managed" streams here instead?

Also using the term "reroute" - That reads to me to mean using the reroute processor, but I think you can get to here via any method of changing the index name.

Maybe less is more here, something like
```suggestion
                                            "pipeline [%s] can't change the target index (from [%s] to stream [%s]) for document [%s]. History: [%s]",
```
Or along those lines? Not sure if we want to say "wired stream" or "managed stream" but I'm assuming those classifications are not easily determined at this point in the system.